### DISCUSSION TITLE
Is my training data invalid?

### DISCUSSION THREAD

**Discussion by dominic-black** [BrainJS/brain.js#827] [Q&A]
Hey guys.
When I call net.run(data)[0] i'm getting undefined as a result. I'm not sure if it is because my training data is in the wrong format. The format is {input: {} {}, output: 0}.

Here is the training data I am trying to test with: https://pastebin.com/qYsuBqG0 .

Any help would be greatly appreciated! 

Thank you1

**Comment by robertleeplummerjr** [comment]
The input data isn't shaped correctly.  The output should be an object or array:
```js
{input: {}, output: [0]}.
```

**Reply by dominic-black** [reply]
Oh i see. This is my new input data: https://pastebin.com/J6Hp7Tas.

I'm now getting Nan instead of undefined. Or when I console log I get {"0":null}. Any ideas?

**Reply by dominic-black** [reply]
So net.run(testData)[0] is always NaN.

**Comment by robertleeplummerjr** [comment]
Your input from that link is as well incorrectly shaped:
```js
let data = [
  {
    input: {
      BlueTeam: {
        Aatrox: 0,
      },
      RedTeam: {
        Aatrox: 1,
```

Input needs to be either an array or flat object who's properties are decimals or integers.  It should be something more like:
```js
let data = [
  {
    input: {
      BlueTeam_Aatrox: 0,
      RedTeam_Aatrox: 1,
  },
```

**Reply by dominic-black** [reply]
I see. So would something like { input: [ [0, 0, 0, 1], [0, 1, 1, 0] ], output: [1] } work?

**Comment by robertleeplummerjr** [comment]
>I see. So would something like { input: [ [0, 0, 0, 1], [0, 1, 1, 0] ], output: [1] } work?

That is as well an alternate that will work.

**Reply by dominic-black** [reply]
But a single array full of objects wouldn't work?

**Reply by dominic-black** [reply]
like { input: [ {a: 0, b: 0, c: 0, d: 1}, {a: 0, b: 1, c: 1, d: 0} ], output: [1] }

**Reply by dominic-black** [reply]
> > I see. So would something like { input: [ [0, 0, 0, 1], [0, 1, 1, 0] ], output: [1] } work?
> 
> That is as well an alternate that will work.

Hey just wondering if i'm doing this right.
My training data looks like this:
`  {
    input: [
      [
        0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      ],
      [
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      ],
    ],
    output: [0],
  },`

and then i'm passing 
`      [
        0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
      ],
      [
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      ],
    ],`

but once again I am getting NaN as a result :(

**Comment by robertleeplummerjr** [comment]
no.

**Reply by dominic-black** [reply]
Okay thanks a lot for your help! I understand how inputs need to work now.

**Reply by dominic-black** [reply]
Is there any issues with having an array of strings input and then a number output? I am getting NaN as a result when I try this.
